doc/api/http.md

```diff
@@ -1539,6 +1539,8 @@ convenience method. The only difference between this method and
 automatically. Note that response data must be consumed in the callback
 for reasons stated in [`http.ClientRequest`][] section.
 
+Note: The `options` properties are copied via `util._extend()` which doesn't include the prototype.
```

Comment:
Aww sorry. 
@jasnell You are right.

@Trott I was thinking adding it to all affected `options` occurrences later on.
I also like the idea to attach it directly to the explanation of `options`. If we can agree I will rebase and push again.